### PR TITLE
[WFCORE-5452] Sync updates to TLS configuration and minor changes for Java 17.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.redhat.com/browse/WFCORE-5452

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2273,7 +2273,7 @@
             </activation>
             <properties>
                 <modular.jdk.args>
-	            --add-opens=java.base/java.security=ALL-UNNAMED
+	            --add-opens=java.base/java.security=ALL-UNNAMED --add-opens=java.base/sun.security.ssl=ALL-UNNAMED
```
Comment (bstansberry) [inline pom.xml:2276]: Can this be a separate line?

Perhaps and xml comment above too that explains what it's there.

But the separate line would improve the odds that future git annotate users will be able to understand why it is there.

File: pom.xml [hunk 2]
```diff
@@ -2272,8 +2272,12 @@
                 <!-- match 9+ -->
             </activation>
             <properties>
+                <!-- 
+                    java.base/sun.security.ssl - Added as TLS configuration within Elytron loads SunJSSE provider directly.
+                -->
                 <modular.jdk.args>
 	            --add-opens=java.base/java.security=ALL-UNNAMED
+	            --add-opens=java.base/sun.security.ssl=ALL-UNNAMED
```
Comment (darranl) [inline pom.xml:2280]: @bstansberry here it is split onto it's own line.

**Review by ropalka — APPROVED** [review]
Verified on JDK17 locally - LGTM

**Comment by jmesnil** [comment]
@darranl thanks

**Pull request merged**